chainer/function.py

```diff
@@ -189,8 +189,10 @@ def __call__(self, *inputs):
         if self.type_check_enable:
             self._check_data_type_forward(in_data)
 
-        hooks = collections.OrderedDict(chainer.get_function_hooks())
-        hooks.update(self.local_function_hooks)
+        hooks = chainer.get_function_hooks()
+        if self.n_local_function_hooks != 0:
```

Comment:
If we use `len(self.n_local_function_hooks)`, a python object is created.
It has overhead.